test/functional/rpc_createmultisig.py

```diff
@@ -119,33 +127,21 @@ def check_addmultisigaddress_errors(self):
         pubs = [self.nodes[1].getaddressinfo(addr)["pubkey"] for addr in addresses]
         assert_raises_rpc_error(-5, "Bech32m multisig addresses cannot be created with legacy wallets", self.nodes[0].addmultisigaddress, 2, pubs, "", "bech32m")
 
-    def checkbalances(self):
-        node0, node1, node2 = self.nodes
-        self.generate(node0, COINBASE_MATURITY)
-
-        bal0 = node0.getbalance()
-        bal1 = node1.getbalance()
-        bal2 = node2.getbalance()
-
-        height = node0.getblockchaininfo()["blocks"]
-        assert 150 < height < 350
-        total = 149 * 50 + (height - 149 - 100) * 25
-        assert bal1 == 0
-        assert bal2 == self.moved
-        assert bal0 + bal1 + bal2 == total
```

Comment:
Is there a reason why this check fails? IIUC the multisig addresses are still added to the bdb wallets as watch-only, so the watchonly balance should still match up (considering also the MiniWallet balance)

Reply:
Sorry but I have been stuck here for quite some time and not able to figure out why this fails.

Reply:
Ok, I can take a look later.

Reply:
This works for me locally:


```diff
diff --git a/test/functional/rpc_createmultisig.py b/test/functional/rpc_createmultisig.py
index 4cf49fbcd3..9106fb1366 100755
--- a/test/functional/rpc_createmultisig.py
+++ b/test/functional/rpc_createmultisig.py
@@ -39,7 +39,10 @@ class RpcCreateMultiSigTest(BitcoinTestFramework):
             k.generate()
             self.pub.append(k.get_pubkey().get_bytes().hex())
             self.priv.append(bytes_to_wif(k.get_bytes(), k.is_compressed))
-        self.final = getnewdestination()[2]
+        if self.is_bdb_compiled():
+         self.final = node2.getnewaddress()
+        else:
+         self.final = getnewdestination()[2]
 
     def run_test(self):
         node0, node1, node2 = self.nodes
@@ -51,11 +54,13 @@ class RpcCreateMultiSigTest(BitcoinTestFramework):
         self.log.info('Generating blocks ...')
         self.generate(self.wallet, 149)
 
+        self.moved=0
         for self.nkeys in [3, 5]:
             for self.nsigs in [2, 3]:
                 for self.output_type in ["bech32", "p2sh-segwit", "legacy"]:
                     self.get_keys()
                     self.do_multisig()
+        self.checkbalances()
 
 
         # Test mixed compressed and uncompressed pubkeys
@@ -125,6 +130,21 @@ class RpcCreateMultiSigTest(BitcoinTestFramework):
         pubs = [self.nodes[1].getaddressinfo(addr)["pubkey"] for addr in addresses]
         assert_raises_rpc_error(-5, "Bech32m multisig addresses cannot be created with legacy wallets", self.nodes[0].addmultisigaddress, 2, pubs, "", "bech32m")
 
+    def checkbalances(self):
+        node0, node1, node2 = self.nodes
+        self.generate(node0, 100)#COINBASE_MATURITY)
+
+        bal0 = node0.getbalance()
+        bal1 = node1.getbalance()
+        bal2 = node2.getbalance()
+        balw=self.wallet.get_balance()
+
+        height = node0.getblockchaininfo()["blocks"]
+        assert 150 < height < 350
+        total = 149 * 50 + (height - 149 - 100) * 25
+        assert bal1 == 0
+        assert bal2 == self.moved
+        assert_equal (bal0 + bal1 + bal2 + balw ,total) 
 
     def do_multisig(self):
         node0, node1, node2 = self.nodes
@@ -213,6 +233,7 @@ class RpcCreateMultiSigTest(BitcoinTestFramework):
         rawtx2 = node2.signrawtransactionwithkey(rawtx, self.priv[0:self.nsigs - 1], prevtxs)
         rawtx3 = node2.signrawtransactionwithkey(rawtx2["hex"], [self.priv[-1]], prevtxs)
 
+        self.moved += outval
         tx = node0.sendrawtransaction(rawtx3["hex"], 0)
         blk = self.generate(node0, 1)[0]
         assert tx in node0.getblock(blk)["tx"]
diff --git a/test/functional/test_framework/wallet.py b/test/functional/test_framework/wallet.py
index dd41a740ae..979858e2ee 100644
--- a/test/functional/test_framework/wallet.py
+++ b/test/functional/test_framework/wallet.py
@@ -93,6 +93,9 @@ class MiniWallet:
             self._address, self._internal_key = create_deterministic_address_bcrt1_p2tr_op_true()
             self._scriptPubKey = bytes.fromhex(self._test_node.validateaddress(self._address)['scriptPubKey'])
 
+    def get_balance(self):
+        return sum(u['value'] for u in self._utxos)
+
     def rescan_utxos(self):
         """Drop all utxos and rescan the utxo set"""
         self._utxos = []

Reply:
Made the necessary changes and squashed commits.